include/af/algorithm.h

```diff
@@ -512,6 +699,37 @@ extern "C" {
     AFAPI af_err af_product_nan(af_array *out, const af_array in, const int dim, const double nanval);
 #endif
 
+#if AF_API_VERSION >= 37
+    /**
+       C Interface for product of elements in an array according to key
+
+       \param[out] keys_out will contain the reduced keys in \p vals along \p dim
+       \param[out] vals_out will contain the product of all values in \p vals along \p dim according to \p keys
+       \param[in] keys is the key array
+       \param[in] vals is the array containing the values to be reduced
+       \param[in] dim The dimension along which the add operation occurs
```

Comment:
"add" should be "product". For the rest of the instances of this typo, please see my comments on their C++ counterparts above.